synapse/storage/databases/main/event_federation.py

```diff
@@ -1179,7 +1179,7 @@ def _get_rooms_with_many_extremities_txn(txn: LoggingTransaction) -> List[str]:
         )
 
     @cached(max_entries=5000, iterable=True)
-    async def get_latest_event_ids_in_room(self, room_id: str) -> Sequence[str]:
+    async def get_latest_event_ids_in_room(self, room_id: str) -> List[str]:
```

Comment:
IIRC we want `@cached` methods to return immutable types, or else callers can accidentally alter the in-cache values. I can't remember how `@cached` works, but I'm pretty sure it'll just use the decorated function's return type without trying to make it immutable.

Suggest `StrSequence` here, or maybe even `StrCollection` since there's no particular ordering to the extremities we select.

Reply:
C.f. https://github.com/matrix-org/synapse/pull/13755

Reply:
and also [`32b4730` (#14870)](https://github.com/matrix-org/synapse/pull/14870/commits/32b473054fc6073c4d5486df027169c8bd58b573)

Reply:
Good-catch. I think you have a branch somewhere which enforces this actually!

Reply:
Aha, https://github.com/matrix-org/synapse/pull/14911. I'd completely forgotten!